index.src.html

```diff
@@ -750,6 +756,130 @@ <h3 id="try-delivery" algorithm>
+  <a>reports</a> from JavaScript when they are created.
+  
+  Issue: How do we specify which reports are observable?  Network-layer reports
+  like CSP reports will not be.  Maybe there's something about the settings
+  object?
+  
+  A <dfn id=dom-report interface>Report</dfn> is the application exposed
+  representation of a <a>report</a>. <dfn attribute for="Report">type</dfn>
+  returns [=report/type=], <dfn attribute for="Report">url</dfn> returns
+  [=report/url=] and <dfn attribute for="Report">body</dfn> returns
+  [=report/body=].
+    
+  Each {{ReportingObserver}} object has these associated concepts:
+    - A <dfn for=ReportingObserver>callback</dfn> set on creation.
+    - An {{ReportingObserverOptions}} dictionary called
+      <dfn for=ReportingObserver>options</dfn>.
+    - A list of {{Report}} objects called the <dfn for=ReportingObserver>report
+      list</dfn> that is initially empty.
+
+  The <dfn constructor for=ReportingObserver><code>
+  ReportingObserver(|callback|)</code></dfn> constructor, when invoked, must
+  create a new {{ReportingObserver}} object with <a>callback</a> set to
+  |callback|, and then return it.
+  
+  The <dfn method for=ReportingObserver><code>observe(|options|)</code></dfn>
+  method, when invoked, must run these steps:
+  
+  1. Set the <a>context object</a>'s <a>options</a> to |options|.
+  
+  2. If it is not already present, add the <a>context object</a> to the 
+     list of <a>registered reporting observers</a> for the <a>environment
+     settings object</a> associated with the <a>context object</a>.
+  
+  The <dfn method for=ReportingObserver><code>disconnect()</code></dfn> method,
+  when invoked, must run these steps:
+  
+  1. If present, remove the <a>context object</a> from the list of
+     <a>registered reporting observers</a> for the <a>environment settings
+     object</a> associated with the <a>context object</a>.
+ 
+  <h3 id="notify-observers" algorithm>
+    Notify reporting observers on |settings| with |report|
+  </h3>
+  
+  This algorithm makes |report|'s contents available to any registered
+  {{ReportingObserver}}s on the provided <a>environment settings object</a>
+  |settings|.
+  
+  1. For each {{ReportingObserver}} |observer| of the <a>registered
+     observers</a> associated with |settings|:
+     
+     1. If |observer|'s <a>options</a> has a {{ReportingObserverOptions/types}}
+        member which does not contain |report|'s [=report/type=], skip these
+        sub-steps.
+        
+     2. Create a new {{Report}} |r| with {{Report/type}} initialized to
+        |report|'s [=report/type=], {{Report/url}} initialized to |report|'s
+        [=report/url=], and {{Report/body}} initialized to |report|'s
+        [=report/body=].
+
+        Issue: how to polymorphically initialize body?
+       
+     3. Add |r| to the end of |observer|'s <a>report list</a>.
+         
+  2. <a>Queue a task</a> to [[#invoke-observers]] with a copy of the
```

Comment:
This should probably be queued as a low priority task? For PerfObserver we define own task source and indicate that it should be a low priority queue: https://w3c.github.io/performance-timeline/#queue-a-performanceentry

Reply:
Thanks, filed https://github.com/WICG/reporting/issues/58 to discuss further.